rust/osm2lanes/src/transform.rs

```diff
@@ -502,34 +512,129 @@ fn osm_separation_type(x: &str) -> Option<BufferType> {
 
 pub fn lanes_to_tags(lanes: &[LaneSpec], _cfg: &Config) -> Result<Tags, LaneSpecError> {
     let mut tags = std::collections::BTreeMap::new();
+    let mut oneway = false;
     tags.insert("highway".to_owned(), "yes".to_owned()); // TODO, what?
+                                                         // Lanes
     {
         let lane_count = lanes
             .iter()
-            .filter(|lane| lane.lane_type == LaneType::Driving)
+            .filter(|lane| {
+                lane.lane_type == LaneType::Driving || lane.lane_type == LaneType::SharedLeftTurn
+            })
             .count();
         tags.insert("lanes".to_owned(), lane_count.to_string());
     }
+    // Oneway
     if lanes
         .iter()
         .filter(|lane| lane.lane_type == LaneType::Driving)
         .all(|lane| lane.direction == Direction::Forward)
     {
+        oneway = true;
         tags.insert("oneway".to_owned(), "yes".to_owned());
     }
-    if lanes.first().unwrap().lane_type == LaneType::Sidewalk
-        && lanes.last().unwrap().lane_type == LaneType::Sidewalk
+    // Pedestrian
     {
-        tags.insert("sidewalk".to_owned(), "both".to_string());
+        match (lanes.first().unwrap().lane_type == LaneType::Sidewalk, lanes.last().unwrap().lane_type == LaneType::Sidewalk)
+        {
+            (false, false) => {}
+            (true, false) => assert!(tags.insert("sidewalk".to_owned(), "left".to_owned())
+                .is_none()),
+            (false, true) => assert!(tags.insert("sidewalk".to_owned(), "right".to_owned())
+                .is_none()),
+            (true, true) => assert!(tags.insert("sidewalk".to_owned(), "both".to_owned())
+                .is_none()),
+        }
+    }
+    // Parking
+    match (
+        lanes
+            .iter()
+            .take_while(|lane| lane.lane_type != LaneType::Driving)
```

Comment:
This is a pretty easy-to-understand way of finding the parking lanes on both sides, I like it. In A/B Street, I tried to give every lane an orientation, including parking lanes, and there were lots of weird bugs -- like sometimes the parking lane not matching the orientation of the driving lane adjacent to it. (I think the bug was with [parking, one-way driving, parking])

But this side-steps all of that; the only thing that would get confusing is [d, p, p, d], but I've _almost_ never seen center parking like this, and OSM can't tag it anyway, so...